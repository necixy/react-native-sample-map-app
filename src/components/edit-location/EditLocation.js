import React, { Component } from "react";
import { View, Text, TextInput, Button } from "react-native";
import { graphql } from "react-apollo";
import locationsQuery from "../../graphql/queries/locations";
import updateLocation from "../../graphql/mutations/updateLocation";
import s from "./EditLocationStyles";

class EditLocation extends Component {
  location;
  static navigationOptions = {
    title: "Edit Location"
  };

  constructor(props) {
    super(props);
    const { navigation } = this.props;
    this.location = navigation.getParam("location");
    this.state = {
      locationName: this.location.name,
      locationAddress: this.location.address
    };
  }

  focusInput(ref) {
    this.refs[ref].focus();
  }

  async updateLocation() {
    const { location } = this;
    await this.props.mutate({
      variables: {
        id: location.id,
        name: this.state.locationName,
        address: this.state.locationAddress,
        lat: location.lat,
        lng: location.lng
      },
      refetchQueries: [{ query: locationsQuery }]
    });
    this.props.navigation.pop();
  }

  render() {
    return (
      <View style={s.container}>
        <View style={s.inputContainer}>
          <TextInput
            testId="inputName"
            placeholder="Name"
            onChangeText={text => this.setState({ locationName: text })}
            value={this.state.locationName}
            returnKeyType="next"
            onSubmitEditing={this.focusInput.bind(this, "inputAddress")}
            style={s.textInput}
          />
          <TextInput
            testId="inputAddress"
            ref="inputAddress"
            placeholder="Address"
            onChangeText={text => this.setState({ locationAddress: text })}
            value={this.state.locationAddress}
            style={s.textInput}
            returnKeyType="done"
          />
        </View>
        <Button title="Update" onPress={this.updateLocation.bind(this)} />
      </View>
    );
  }
}

export default graphql(updateLocation)(EditLocation);
