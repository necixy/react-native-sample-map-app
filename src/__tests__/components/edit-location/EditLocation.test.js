import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";
import EditLocation from "../../../components/edit-location/EditLocation";

const location = {
  name: "Berlin",
  address: "Berlin, Germany"
};
const mockedNavigation = {
  getParam: paramName => {
    switch (paramName) {
      case "location":
        return location;

      default:
        break;
    }
  }
};

describe("EditLocation Component Testing", () => {
  it("EditLocation Renders correctly", async () => {
    const component = renderer.create(
      <MockedProvider addTypename={false}>
        <EditLocation navigation={mockedNavigation} />
      </MockedProvider>
    );

    let root = component.root;
    let editLocationInstance =
      root.children[0].children[0].children[0].children[0].instance;
    const { locationName, locationAddress } = editLocationInstance.state;

    const inputName = root.find(
      el => el.props && el.props.testId == "inputName" && el.type == "TextInput"
    );

    const inputAddress = root.find(
      el =>
        el.props && el.props.testId == "inputAddress" && el.type == "TextInput"
    );

    expect(inputName.props.value).toBe(locationName);
    expect(inputAddress.props.value).toBe(locationAddress);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
