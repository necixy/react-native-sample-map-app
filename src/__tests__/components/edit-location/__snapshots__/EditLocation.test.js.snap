// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditLocation Component Testing EditLocation Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Name"
      returnKeyType="next"
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 1,
          "height": 50,
          "marginBottom": 10,
          "padding": 10,
          "width": "100%",
        }
      }
      testId="inputName"
      underlineColorAndroid="transparent"
      value="Berlin"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Address"
      returnKeyType="done"
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 1,
          "height": 50,
          "marginBottom": 10,
          "padding": 10,
          "width": "100%",
        }
      }
      testId="inputAddress"
      underlineColorAndroid="transparent"
      value="Berlin, Germany"
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Update
      </Text>
    </View>
  </View>
</View>
`;
